File: src/main/java/net/Chidoziealways/everythingjapanese/entity/client/ya/YaRenderState.java
package net.Chidoziealways.everythingjapanese.entity.client.ya;

import net.Chidoziealways.everythingjapanese.entity.custom.YaProjectileEntity;
import net.minecraft.client.renderer.entity.state.EntityRenderState;

public class YaRenderState extends EntityRenderState {
    private float partialTicks;
    private YaProjectileEntity entity;

    public float getPartialTicks() {
        return partialTicks;
    }

    public void setPartialTicks(float partialTicks) {
        this.partialTicks = partialTicks;
    }

    public YaProjectileEntity getEntity() {
        return entity;
    }

    public void setEntity(YaProjectileEntity entity) {
        this.entity = entity;
    }
}

File: src/main/kotlin/net/Chidoziealways/everythingjapanese/entity/client/ya/YaRenderState.kt
package net.Chidoziealways.everythingjapanese.entity.client.ya

import net.Chidoziealways.everythingjapanese.entity.custom.YaProjectileEntity
import net.minecraft.client.renderer.entity.state.EntityRenderState

class YaRenderState : EntityRenderState() {
    private var partialTicks = 0f
    private var entity: YaProjectileEntity? = null

    fun getPartialTicks(): Float {
        return partialTicks
    }

    fun setPartialTicks(partialTicks: Float) {
        this.partialTicks = partialTicks
    }

    fun getEntity(): YaProjectileEntity? {
        return entity
    }

    fun setEntity(entity: YaProjectileEntity?) {
        this.entity = entity
    }
}
